Refactors all the Post-related classes to remove wrapper divs, remove legacy stuff, and split up huge classes

essential-randomness opened the pull request.

virtuous-cat commented on tests/05-Posts/03-PostFooter.test.tsx:
```diff
@@ -0,0 +1,200 @@
+    largeContainer = screen.getByTestId("large-container");
+
+    expect(WithNotes.args!.allowsContribution).toBeTrue();
+    expect(WithNotes.args!.allowsComment).toBeTrue();
```
I think this should be 
```
    expect(Answerable.args!.allowsContribution).toBeTrue();
    expect(Answerable.args!.allowsComment).toBeTrue();
```

essential-randomness commented on src/buttons/Button.tsx:
```diff
@@ -114,6 +114,9 @@ const Button: React.FC<ButtonProps> = ({
         }
+        .button.disabled {
+          opacity: 50%;
+        }
```
It probably needs to be better than this, but I decided to not have perfect be the enemy of the good. I'm going to update that issue saying I hacked together something but it's still needed after this is checked in.

essential-randomness commented on src/post/Avatar.tsx:
```diff
@@ -131,7 +155,7 @@ export interface AvatarProps {
   userIdentity?: UserIdentityType;
   forceHide?: boolean;
-  compact?: boolean;
+  compact?: boolean | "mini";
```
I actually didn't mean to keep this as a long-term thing, but I added a comment to clarify what my ideal solutions are (as I should have done before) :P

essential-randomness commented on src/post/Footer.tsx:
```diff
@@ -101,96 +196,40 @@ const Footer: React.FC<FooterProps> = ({
+            compact={currentBreakpoint == "compact"}
+            theme={ButtonStyle.TRANSPARENT}
+            disabled={!allowsComment}
+            label="new comment"
```
I changed them, but I wish we had an accessibility expert to ask. I feel like the visuals make it clearer what the action is, and without those it might be harder unless you're told that it's for a "new comment/contribution". But I don't know enough to be sure.

virtuous-cat replied:
Since they are buttons, screenreaders are supposed to announce that them as "{label} button" and I'm pretty sure you're supposed to favor verbs for labeling them. I *think* "contribute button" and "comment button" are about as clear auditorily as they are visibly, but if further clarity was needed I think "add contribution/comment" would be preferable to "new contribution/comment". But, best practices is actually for them to be labelled by the visual text content, so that's why I think they should match it, since it does exist at the larger screen sizes.

essential-randomness replied:
Thank you for the explanation, I learned things! It should be good as it is then :)

essential-randomness commented on src/post/Post.tsx:
```diff
@@ -7,137 +8,134 @@ import {
+  .metadata {
+    max-width: calc(100% - 20px);
+  }
+  button {
```
Very fair. OTOH I don't think this is always true, for example "header/footer" in the cards I feel is clear enough (because usually you always have one). But I very much agree with the general principle.

It would be nice for us to come up with guidelines for this, but maybe let's give it a little bit to see how we feel about doing stuff this way in the future.

FWIW, I could put this before or after, I haven't actually figured out what the right position is.

essential-randomness commented on src/post/Post.tsx:
```diff
@@ -425,46 +218,40 @@ export interface PostHandler {
-  hideFooter?: boolean;
-  hideUpdates?: boolean;
   backgroundColor?: string;
+  muted?: boolean;
```
Maybe, I don't know. The thought was that you can't actually mute a single post right now, but I do think you're right, so I moved it.

essential-randomness commented:
Addressed comments and did a couple fixes/changes. Most importantly: I made the Thread story work again (the issue was that it was using an old format for the comment thread hidden within, which no longer works. 

The Big Q: its style seems to match the other in the file. Should I update it to the new story format, or is that compatible? I can update them all if needed.

Also, let's get rid of the deprecated thread formats (in a different PR).

virtuous-cat reviewed (commented):
> Addressed comments and did a couple fixes/changes. Most importantly: I made the Thread story work again (the issue was that it was using an old format for the comment thread hidden within, which no longer works.
> 
> The Big Q: its style seems to match the other in the file. Should I update it to the new story format, or is that compatible? I can update them all if needed.

One of my todos is updating all the stories to the new format, so it's going happen one way of the other. If you want to do these ones now while it's fresh in your mind that's great, but otherwise I will do them when I do the rest.

essential-randomness commented on src/post/CompactPostThread.tsx:
```diff
@@ -104,7 +104,7 @@ const PreviousContent = (props: CompactThreadProps) => {
                 newContributions={post.newContributions}
                 newComments={post.newComments}
-                notesLink={post.notesLink}
+                notesLink={post.createdTimeLink}
```
I'm open for suggestions. It's possible that we just need to add more linking options for CompactPost, since it's possible that consumers will want the intermediate "notes" to have a different interaction than the ones at the end (right now they aren't even that visually distinctive, but they should be more so later).

I think the issue with naming them based on what they do is that I assume it's legit to make different choices there in terms of views/clients.

essential-randomness commented:
> > Addressed comments and did a couple fixes/changes. Most importantly: I made the Thread story work again (the issue was that it was using an old format for the comment thread hidden within, which no longer works.
> > The Big Q: its style seems to match the other in the file. Should I update it to the new story format, or is that compatible? I can update them all if needed.
> 
> One of my todos is updating all the stories to the new format, so it's going happen one way of the other. If you want to do these ones now while it's fresh in your mind that's great, but otherwise I will do them when I do the rest.

I've done it a little bit (it seems like the format they're in now is compatible with the story format, but I made the names a bit better), but these stories are a mess and I hope to get to do some work on threads refactoring too, so maybe I can give them a better attempt at clean up later.

virtuous-cat reviewed (approved):
Just one comment to maybe add, otherwise I don't have anything else

essential-randomness commented:
I went and just added the intermediate link. Thank you so much for your review!
